packages/clerk-js/src/core/auth/AuthCookieService.ts

```diff
@@ -52,11 +52,16 @@ export class AuthCookieService {
   ) {
     // set cookie on token update
     eventBus.on(events.TokenUpdate, ({ token }) => {
+      // only update session cookie from active tab
+      if (!document.hasFocus()) {
+        return;
+      }
```

Comment:
We might want more nuance here, e.g. checking if there is only a single tab open (even if in the background). The token is already refreshed on refocus of the tab, so the session cookie should quickly get refreshed.

Direct access of the cookie is discouraged, so in practice this _shouldn't_ be an issue...

Reply:
☁️ I think we should not add this guard here since it will disable updating the cookie on background which may result in increasing the HTTP 401 errors shown in customers due to requests triggered in background of customers applications.
Also wouldn't we want to have the session updated on the following cases:
- different multiple browser windows open at the same time (or using split screen)
- having the tab open without having the document focused
- having the developer tools open (`document.hasFocus()` returns false)

Reply:
@dimkl Right, I think the check needs to be more robust, but I think we do need to include something. The issue is that two tabs / windows with different selected orgs would be updating the cookie at the same time. So the org ID in the cookie would not be consistent.